xrmocap/core/evaluation/fourd_association_evaluation.py

```diff
@@ -0,0 +1,440 @@
+# yapf: disable
+import csv
+import logging
+import numpy as np
+import os.path as osp
+import time
+from prettytable import PrettyTable
+from tqdm import tqdm
+from typing import List, Tuple, Union
+from xrprimer.utils.log_utils import get_logger
+from xrprimer.utils.path_utils import prepare_output_path
+
+from xrmocap.data.data_visualization.builder import (
+    BaseDataVisualization, build_data_visualization,
+)
+from xrmocap.data.dataset.builder import MviewMpersonDataset, build_dataset
+from xrmocap.data_structure.keypoints import Keypoints
+
+from xrmocap.ops.fourd_association.builder import (
+    FourdAssociator, build_fourd_associator
+)
+from xrmocap.transform.convention.keypoints_convention import (
+    convert_keypoints, get_keypoint_idx, get_keypoint_num,
+)
+from xrmocap.transform.limbs import get_limbs_from_keypoints
+from xrmocap.utils.geometry import compute_similarity_transform
+from xrmocap.utils.mvpose_utils import (
+    add_campus_jaw_headtop, add_campus_jaw_headtop_mask, check_limb_is_correct,
+    compute_mpjpe, vectorize_distance,
+)
+
+# yapf: enable
+# skel19_info = {'skel19':range(19)}
+
+class FourDAssociationEvaluation:
+
+    def __init__(self,
+                 output_dir: str,
+                 selected_limbs_name: List[List[str]],
+                 additional_limbs_names: List[List[str]],
+                 dataset: Union[dict, MviewMpersonDataset],
+                 associator: Union[dict, FourdAssociator],
+                 dataset_visualization: Union[None, dict,
+                                              BaseDataVisualization] = None,
+                 pred_kps3d_convention: str = 'coco',
+                 eval_kps3d_convention: str = 'campus',
+                 logger: Union[None, str, logging.Logger] = None) -> None:
+        """Top-down association evaluation.
+
+        Args:
+            output_dir (str): The path to save results.
+            selected_limbs_name (List[List[str]]): The name of selected
+                limbs in evaluation.
+            additional_limbs_names (List[List[str]]):
+                Names at both ends of the limb.
+            dataset (Union[dict, MviewMpersonDataset])
+            associator (Union[dict, MvposeAssociator])
+            dataset_visualization
+                (Union[None, dict, BaseDataVisualization], optional):
+                Defaults to None.
+            pred_kps3d_convention (str, optional): Target convention of
+                keypoints3d, Defaults to 'coco'.
+            eval_kps3d_convention (str, optional): the convention of
+                keypoints3d for evaluation, Defaults to 'campus'.
+            logger (Union[None, str, logging.Logger], optional):
+                Logger for logging. If None, root logger will be selected.
+                Defaults to None.
+        """
+
+        self.output_dir = output_dir
+        self.pred_kps3d_convention = pred_kps3d_convention
```

Comment:
Convention of the output kps3d is determined by fourd_associator, can we get it from the associator instead of setting it manually? Assume that we pass `pred_kps3d_convention=smplx` manually, will there be an error?

Reply:
Since the convention (Skel19) used in 4-D isn't defined in mmhuman3d, I manually convert the Skel19 convention to Campus in `associator`. So the 4-D method currently only supports prediction in Campus convention.
I try to add new convention in `keypoints_convention`.